app/src/main/java/com/eveningoutpost/dexdrip/Services/LibreWifiReader.java

```diff
@@ -236,7 +315,9 @@ private static boolean hasGeoLocation(LibreWifiData record) {
                     && (Home.get_engineering_mode())
                     && (DexCollectionType.getDexCollectionType() == DexCollectionType.Mock))) {
                 tmpList = readFake();
-            } else {
+            } else if ( host.startsWith("https://")) {
+                tmpList = readHttpJson(host, numberOfRecords);
+            }else {
                 tmpList = ReadHost(host, numberOfRecords);
```

Comment:
Which protocol could that be?

Reply:
There is a protocol between xDrip and the pis based on json.
Used for direct reading of data.

Reply:
Please can you provide a link to this protocol? Just for my understanding.

Reply:
The communication is based on TCP/ip connection.
The initiatior sends a request with ComunicationHeaderV2 

class ComunicationHeaderV2 {
    @Expose
    int version;
    @Expose
    int numberOfRecords;
    
    // Only send packets that are newer than this time
    @Expose
    long fromTime;
    
    // bluetooth device addresses
    @Expose
    String btAddresses;
}


And the reciever sends a LibreWifiHeader with the results.
class LibreWifiHeader {
    // Version of the reply
    @Expose
    int reply_version;
    
    // Maximum version of protocol supported by this version
    @Expose
    int max_protocol_version;
    
    // Time of last reading from libre (can be no sensor for example
    @Expose
    long last_reading;
    
    // Any debug message that wants to be displayed.
    @Expose
    String debug_message;
    
    @Expose
    String device_type;
    
    // The actual data
    @Expose
    LibreWifiData [] libre_wifi_data;

    @Override
}

public class LibreWifiData {
    
    @Expose
    String BlockBytes; // The base 64 encoded FARM
    
    @Expose
    Long CaptureDateTime;
    
    @Expose
    public long RelativeTime;
    
    @Expose
    int ChecksumOk ;
    
    @Expose
    String DebugInfo;
    
    @Expose
    int TomatoBatteryLife;
    
    @Expose
    int UploaderBatteryLife;
    
    @Expose
    int Uploaded;
    
    @Expose
    String HwVersion;
    
    @Expose
    String FwVersion;
    
    @Expose
    String SensorId;
    
    @Expose
    String patchUid; // The base 64 encoded patchUid
    
    @Expose
    String patchInfo; // The base 64 encoded patchInfo
    
    public Long getCaptureDateTime() {
        return CaptureDateTime;
    }
   
 }